### PR TITLE
[DOCFIX] Fix configuration setting documentation to prefer site prope…

### PR DESCRIPTION
**Pull request by apc999**
…rties

### CODE REVIEW

**Comment by calvinjia** [comment]
alluxio-bot, check this please.

**Comment by calvinjia** [comment]
LGTM, I think there are still a few occurrences of `alluxio-env` (ie. Running HadoopMR/Spark on Alluxio)

**Comment by calvinjia** [comment]
@apc999 I think there's also a mention of setting an `alluxio-env` option in Configuring-with-S3 (the server side encryption part)

**Comment by apc999** [comment]
@calvinjia Fixed a bunch (and more to clarify the configuration setting in general). PTAL

File: docs/_includes/Common-Commands/copy-alluxio-site-properties.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 ```bash
-$ cp $ALLUXIO_HOME/conf/alluxio-site.properties.template ~/.alluxio/alluxio-site.properties
-```
\ No newline at end of file
+$ cp $ALLUXIO_HOME/conf/alluxio-site.properties.template /etc/alluxio/alluxio-site.properties
+```
```
Comment (calvinjia) [inline docs/_includes/Common-Commands/copy-alluxio-site-properties.md]: The user may not have access to `/etc/`, perhaps we should have both putting it in the conf directory and `/etc/` and say `/etc/` will take effect even in nonAlluxio processes?

File: docs/en/Configuration-Settings.md [hunk 2]
```diff
@@ -45,8 +45,13 @@ Note that, setting Alluxio configuration in this way is application specific and
 
 ## Environment variables
 
-There are a few basic and very frequently used Alluxio configuration properties that can be set via the
-following environment variables:
+> Environment variables are useful to customize configuration of Alluxio when you  
+> [launch Alluxio servers](Running-Alluxio-Locally.html), or [using Alluxio command line interfaces](Command-Line-Interface.html),
+> however, they will not be respected by applications like Spark or MapReduce that use Alluxio as a client.
```
Comment (calvinjia) [inline docs/en/Configuration-Settings.md]: It should work as long as the environment variable is set in the env before the process is launched/

File: docs/en/Configuration-Settings.md [hunk 3]
```diff
@@ -104,13 +109,11 @@ Alternatively, you can create one from a template we provided in the source code
 
 {% include Common-Commands/copy-alluxio-env.md %}
 
+## Property files
 
-Note that `conf/alluxio-env.sh` is sourced when you
-[launch Alluxio servers](Running-Alluxio-Locally.html), or [use Alluxio command line interfaces](Command-Line-Interface.html),
-but not for applications.
-
+> Alluxio property files need to be put on the application JVM classpath, or the pre-defined paths to take effect.
```
Comment (calvinjia) [inline docs/en/Configuration-Settings.md]: Do we want to be more specific and say the parent folders of the conf files?

File: docs/_includes/Common-Commands/copy-alluxio-site-properties.md (hunk 1 above)
Comment (apc999) [inline docs/_includes/Common-Commands/copy-alluxio-site-properties.md]: refactored a bunch, 
see
`en/Configuration-Settings.md`

File: docs/en/Configuration-Settings.md (hunk 2 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: are you asking if setting environment variable works if they are set before a Spark process launched? The answer is no. To make alluxio environment variables work, a process needs to source `libexec/alluxio-config.sh` to translate the   environment variable to Alluxio property opts.

if you are asking whether we need to set environment variable before launching Alluxio processes, the answer is yes.  I updated the doc to make this clear.

File: docs/en/Configuration-Settings.md (hunk 3 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: done.

**Comment by apc999** [comment]
@calvinjia  I refactored configuration doc a bit to address your comments but also make the description more clear/accurate. PTAL

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
Jenkins, test this please.

File: docs/en/Configuration-Settings.md [hunk 4]
```diff
@@ -104,30 +110,38 @@ Alternatively, you can create one from a template we provided in the source code
...
-`alluxio-site.properties` file. On startup, Alluxio checks if the configuration
-property file exists and if so, it uses the content to override the default values of configuration properties.
-This property file is searched in `${HOME}/.alluxio/`, `/etc/alluxio/` (can be customized by changing the default value
-of `alluxio.site.conf.dir`) and the classpath of the Java VM (in
-which Alluxio is running) in order.
-
-For example, one can copy the site properties template in directory
-`${ALLUXIO_HOME}/conf` to `${HOME}/.alluxio/` and edit it to fit your configuration tuning needs.
+For example, `${ALLUXIO_HOME}/conf/` is by default on the classpath of Alluxio master, worker and shell JVM processes.
+So you can simply create `${ALLUXIO_HOME}/conf/alluxio-site.properties` and customize it to fit your configuration tuning needs
```
Comment (gpang) [inline docs/en/Configuration-Settings.md]: should you mention the `conf/alluxio-site.properties.template` file?

**Comment by gpang** [comment]
@apc999 I left a minor comment. Thanks!

File: docs/en/Configuring-Alluxio-with-S3.md [hunk 5]
```diff
@@ -65,20 +65,24 @@ If you are using s3a, you can specify credentials in 4 ways, from highest to low
 See [Amazon's documentation](http://docs.aws.amazon.com/java-sdk/latest/developer-guide/credentials.html#id6)
 for more details.
 
+Alternatively, these configuration settings can be set in the `conf/alluxio-env.sh` file. More
```
Comment (calvinjia) [inline docs/en/Configuring-Alluxio-with-S3.md]: `alluxio-site.properties`?

**Comment by calvinjia** [comment]
LGTM

File: docs/en/Configuration-Settings.md (hunk 4 above)
Comment (apc999) [inline docs/en/Configuration-Settings.md]: fixed. PTAL

**Comment by apc999** [comment]
@gpang  PTAL

**Comment by gpang** [comment]
LGTM

**Pull request merged**